src/main/scala/com/microsoft/partnercatalyst/fortis/spark/transforms/topic/KeywordExtractor.scala

```diff
@@ -0,0 +1,38 @@
+package com.microsoft.partnercatalyst.fortis.spark.transforms.topic
+
+import com.microsoft.partnercatalyst.fortis.spark.transforms.Tag
+import org.apache.commons.collections4.trie.PatriciaTrie
+
+import scala.collection.mutable.ListBuffer
+
+@SerialVersionUID(100L)
+class KeywordExtractor(keyWords: Seq[String]) extends Serializable {
+  private val wordTokenizer = """\b""".r
+  private val keywordTrie = new PatriciaTrie[Unit]()
+  private val originalCase = Map[String, String](keyWords.map(s => (s.toLowerCase, s)): _*)
+
+  originalCase.keys.foreach(keywordTrie.put(_, ()))
+
+  def extractKeywords(text: String): List[Tag] = {
+
+    def findMatches(segment: Seq[String]): Iterable[String] = {
```

Comment:
That's a note for others. It's something I learned while reviewing this code so I figured it'd be good to share it.